builtin/common/tests/vector_spec.lua

```diff
@@ -92,15 +92,67 @@ describe("vector", function()
 			rotated3 = vector.rotate_around_axis(rotated3, axis3, -math.pi / 2)
 			assert.True(almost_equal(rotate3, rotated3))
 		end)
+		it("is_left_handed", function()
+			local v_before1 = {x = 0, y = 1, z = -1}
+			local v_after1 = vector.rotate_around_axis(v_before1, {x = 1, y = 0, z = 0}, math.pi / 4)
+			assert.True(almost_equal(vector.normalize(vector.cross(v_after1, v_before1)), {x = 1, y = 0, z = 0}))
```

Comment:
:thinking: `v1, v2, vector.cross(v1, v2)` are a left-handed system (eg. if you insert (1, 0, 0) for v1 and (0, 1, 0) for v1 into vector.cross, you'll get (0, 0, 1), which is left-handed in minetest).
Aren't `v_after1` and `v_before1` in .cross in the wrong order here? Trying it with my fingers, it shows me that it was a right-handed rotation.

Reply:
> `v1, v2, vector.cross(v1, v2) are a left-handed system`

The cross product is right handed. By putting v_after1 and v_before1 in this order I however get a cross product for a left handed coordinate system. See [this](https://upload.wikimedia.org/wikipedia/commons/thumb/b/b0/Cross_product_vector.svg/800px-Cross_product_vector.svg.png) grafic.

The reason the cross product appears left handed in minetest is convention. I can't however write tests based on that, because left handed rotations would appear right handed and vice versa.

I might be getting something wrong or missing something though.

Reply:
The graphic you linked uses a right-handed system. [Minetest's `vector.cross`](https://github.com/minetest/minetest/blob/27d611fe5561db20b380a16fdc6bcf1fefaf5d39/builtin/common/vector.lua#L84-L90) does the same calculations as the "normal" cross product. As minetest uses a left-handed system, the result will be left-handed in minetest (in the game, for players).
A left-handed system is mirrored to a right-handed one. A right-handed rotation will look like a left-handed rotation in the mirror.

You could also test this manually in game by making an entity that rotates itself pi/4 around some axis when you rightclick it.

Reply:
I've made a small mod for testing and can confirm, that it's right-handed:
```lua
local particle_pos = vector.new(1, 1, 0)
local axis = vector.new(0, 1, 0)
local radians_per_sec = math.pi

minetest.register_globalstep(function(dtime)
	local player = minetest.get_connected_players()[1]
	if not player then
		return
	end

	minetest.add_particle({
		pos = vector.add(particle_pos, player:get_pos()),
		velocity = {x=0, y=0, z=0},
		acceleration = {x=0, y=0, z=0},
		expirationtime = dtime * 1.5,
		size = 1,
		collisiondetection = false,
		collision_removal = false,
		object_collision = false,
		vertical = false,
		texture = "bubble.png",
	})

	particle_pos = vector.rotate_around_axis(particle_pos, axis, radians_per_sec * dtime)
end)
```

Reply:
Thanks, I'm pretty sure I get it now. I think I got confused because in the beginning I implemented vector.forward_at_rotation and vector.up_at_rotation using vector.rotate_around_axis. But since euler angles in MT are counterclockwise I got right looking results with the wrong code.

I guess it's time to fix this now.